### PR TITLE
validate new password

### PR DESCRIPTION
**Pull request by Ugizwenayo-Divine**
#### What does this PR do?
- validate a new password before resetting the password

#### Description of Task to be completed?
- created validatePassword function in middlewares/validation.js file
- created passwordResetValidation function in middlewares/authValidation.js
- called passwordResetValidation middleware before updating password

#### How should this be manually tested?
- After cloning the project repo, run `npm i` inside the cloned repo folder
- Make sure to provide all values for variables under `#Development database`
- Finally, run the following commands respectively:
`git pull`
`git checkout bg-validate-updated-password`
`npx sequelize-cli db:create`
`npx sequelize-cli db:migrate`
- make sure you have some data in users table
- npm run dev
##### In postman
- run `http://localhost:3000/api/auth/resetpassword` 
- then `http://localhost:3000/api/auth/resetpassword/token`
 provide an invalid password, you must get "Password should consists of alphabets, numbers and at least 8 characters minimum" message
#### Any background context you want to provide?
- make sure node is installed on your machine

#### What are the relevant pivotal tracker stories?
N/A

#### Screenshots (if appropriate)
Capture3
Capture4
Capture5
Capture6
Capture7
Capture8

### CODE REVIEW

**Review by muchristian — APPROVED** [review]
you did a great job, but you still have to fix code climate issue

File: src/middlewares/authValidation.js [hunk 1]
```diff
@@ -13,11 +9,23 @@ const { errorResponse } = responseHandlers;
...
 };
 
-export default { signupValidation };
+/**
+ * @param {object} req
+ * @param {object} res
+ * @param {object} next
+ * @returns{object} return next() if validations pass
+ */
+const passwordResetValidation = async (req, res, next) => {
```
Comment (descholar-ceo) [inline src/middlewares/authValidation.js]: I think there is no point of creating the second validation function, you should use the existing validation function which was created to validate signup, and then you edit that function to make it adapt to your needs

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
You did well @Ugizwenayo-Divine but I have a little suggestion for you, try to edit the existing function of validation which was already there, to create a new one I don't think it as a good idea

**Review by Muhire-Josue — COMMENTED** [review]
Can you include screenshots of `responses` from postman, especially error messages. you can add those screenshots in your PR under `Screenshots (if appropriate)`

File: tests/authentication.test.js [hunk 2]
```diff
@@ -321,7 +335,7 @@ describe('Reset Email', () => {
   // occured an error while updating
   it('errored update of the password ', (done) => {
     const pass = {
-      password: 'sesese'
+      password: 'segsese145'
```
Comment (Muhire-Josue) [inline tests/authentication.test.js]: This is very "plain text", can you create a `mockData` for password in `tests/data/mockData.js`

**Review by Muhire-Josue — COMMENTED** [review]
Could you include a label on this PR?

**Review by descholar-ceo — APPROVED** [review]

**Comment by the22mastermind** [comment]
@Ugizwenayo-Divine You did a great job. I would like to suggest you include a middleware that checks token validity on this route `/api/auth/resetpassword/token`, that way when someone takes longer to reset their password and the token expires, they will receive an appropriate message.
Also, you should validate the email, because there's a bug when you submit a empty request i.e email.toLowerCase throws an error
Keep it up.

**Review by Muhire-Josue — COMMENTED** [review]
LGTM

File: src/middlewares/validations.js [hunk 3]
```diff
@@ -34,14 +36,75 @@ const validateSignup = user => {
...
+* @param {object} req
+* @param {object} res
+* @param {object} next
+* @returns {object} error or next
+* @description Verifies if token has not expired
+*/
+export const verifyToken = async (req, res, next) => {
+  try {
+      const { token } = req.params;
+      // jwt.verify(token, process.env.JWT_KEY);
```
Comment (mifeille) [inline src/middlewares/validations.js]: Please remove this commented line.

**Pull request merged**